lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -693,18 +693,25 @@ sub _create_clones ($self, $jobs, $comments, $comment_text, $comment_user_id, @c
         $res->register_assets_from_settings;
     }
 
+    my %git_clones;
+    my @clone_ids;
     for my $original_job_id (@original_job_ids) {
         my $cloned_job = $clones{$original_job_id};
         # calculate blocked_by
         $cloned_job->calculate_blocked_by;
         # add a reference to the clone within $jobs
-        $jobs->{$original_job_id}->{clone} = $cloned_job->id;
+        push @clone_ids, $jobs->{$original_job_id}->{clone} = $cloned_job->id;
+        # add Git repositories to clone
+        create_git_clone_list($cloned_job->settings_hash, \%git_clones);
     }
 
     # create comments on original jobs
     $result_source->schema->resultset('Comments')
       ->create_for_jobs(\@original_job_ids, $comment_text, $comment_user_id, $comments)
       if defined $comment_text;
+
+    # enqueue Minion jobs to clone required Git repositories
+    OpenQA::App->singleton->gru->enqueue_git_clones(\%clones, \@clone_ids);
```

Comment:
This would create a minion job for every job, only cluster jobs would be grouped together, right?
That would be way too many jobs in some cases.
The detection for identical git_clone tasks is not perfect, especially if they are created quickly after each other.

Reply:
Otherwise we really needed a "preparing" state.

I suppose it would work like this:

1. We create jobs and now the initial state is "preparing" instead of "scheduled" keeping track of the job IDs.
2. We enqueue Minion jobs.
3. We set all jobs we haven't created Minion jobs for in step 2 to "scheduled" immediately.
4. Before deleting GRU tasks we set related jobs to "preparing". (This would still not fix the scheduling problem when there's a different set of e.g. download jobs within a parallel cluster - unless we take job dependencies into account here.)

Step 2 and 3 would happen in one transaction. Step 4 would also happen in one transaction.

Reply:
For the sake of simplicity we could also reduce the number of Minion jobs on job restarts by avoiding Minion jobs for `git_auto_update`. Of course then `git_auto_update` would rely only on the periodic updates for restarted jobs.

Reply:
I don't understand step 4. Maybe you can explain tomorrow after the daily?

OTOH we could try this PR and see how it works out in practice.

Theoretically we could find out if any new minion job is required by keeping track of the git directories in the `%git_clones` hash while iterating over the jobs to restart. I guess in most cases we will only have one CASEDIR/NEEDLES_DIR or DISTRI.
Just that we would need to pass an additional `%git_clones_not_yet_enqueued` hash through from the toplevel job_restart, which doesn't sound nice.

Reply:
> For the sake of simplicity we could also reduce the number of Minion jobs on job restarts by avoiding Minion jobs for `git_auto_update`. Of course then `git_auto_update` would rely only on the periodic updates for restarted jobs.

That was a requirement though. We did have a complain from someone who restarted a job and was wondering why it wasn't using the updated code.

Reply:
btw, I guess this should be:
```suggestion
    OpenQA::App->singleton->gru->enqueue_git_clones(\%git_clones, \@clone_ids);
```
Probably the reason for the test failures